### PR TITLE
socket server

### PR DESCRIPTION
**Pull request by joshherkness**

### CODE REVIEW

**Comment by joshherkness** [comment]
I'm currently getting an error when I attempt to create a new session. I believe it stems from the line:

```
somesockets.update(self.game_map_id, self.code)
```

When I comment it out, the error goes away. It says:

```
Error in decorators [decorators.py:26]:
name 'user' is not defined
```

**Comment by joshherkness** [comment]
Looks like the error is stemming from `emit('update', json=map, room=room)`, couldn't figure it out.

File: server/somesockets.py [hunk 1]
```diff
@@ -1,47 +1,87 @@
...
-def handle_RoomNotFound(json):
-    emit('RoomNotFound', json)
-    print ('Room Not Found!')
+def update(map_id, room):
+    """
+    This isn't a socket event, this is triggered from within the flask application.
+    Nobody should be hitting this endpoint.
+    """
+    map = models.GameMap.objects(id=map_id).first()
+    emit('update', json=map, room=room)
```
Comment (AnthonyHewins) [inline server/somesockets.py]: im an idiot i forgot to wrap the map_id in a dict. that might be one of a few issues now that I realize it

Comment (AnthonyHewins) [reply server/somesockets.py]: @joshherkness i will make some updates and then you should try to see if they make a difference within the next hour or so

**Review by AnthonyHewins — APPROVED** [review]
This is as far as I could get it. It says i need to add HTTP context vars to do the last test and I may not be able to even do it without having a real HTTP connection, so the client might be necessary to do the tests completely. Everything seems to work though.

**Comment by AnthonyHewins** [comment]
I'll wait to delete the branch in case you want to make any other changes

**Comment by joshherkness** [comment]
Looks like it's all working properly now. I'm going to delete this branch.

**Pull request merged**